frontend/model/contracts/group.js

```diff
@@ -262,30 +272,41 @@ DefineContract({
       }
     },
     'gi.contracts/group/proposal': {
-      validate: objectOf({
-        proposalType: proposalType,
-        proposalData: object, // data for Vue widgets
-        votingRule: ruleType,
-        expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
-      }),
-      process ({ data, meta, hash }, { state }) {
-        // make sure this isn't a duplicate proposal
+      validate: (data, { state, meta }) => {
+        objectOf({
+          proposalType: proposalType,
+          proposalData: object, // data for Vue widgets
+          votingRule: ruleType,
+          expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
+        })(data)
+
+        // Validate this isn't a duplicate proposal
         for (const hash in state.proposals) {
           const prop = state.proposals[hash]
-          if (prop.status === STATUS_OPEN &&
-            prop.data.proposalType === data.proposalType &&
-            prop.meta.username === meta.username &&
+          if (prop.status !== STATUS_OPEN || prop.data.proposalType !== data.proposalType) {
+            continue
+          }
+
+          if (prop.data.proposalType === PROPOSAL_REMOVE_MEMBER &&
+            prop.data.proposalData.member === data.member
+          ) {
+            throw new TypeError('There is an open proposal to remove them.')
+          }
+
+          if (prop.meta.username === meta.username &&
             deepEqualJSONType(prop.data.proposalData, data.proposalData)
           ) {
-            throw new TypeError(`proposal: is identical to proposal ${hash}`)
+            throw new TypeError('This proposal is identical to another one.')
           }
         }
+      },
```

Comment:
A few notes:

- Both here and in several other places in this file (do a search for `TypeError('`), I forgot to wrap the error messages in `L` functions. Please add them so that the errors get translated.
- We want this function to be as generic as possible. I don't remember why I added the check for `prop.meta.username === meta.username`, but if we remove that then I think it's possible to remove the proposal-specific check on lines 290-294, and instead throw an exception for an "identical proposal" if `deepEqualJSONType(prop.data, data)` ... although double-check me on that (I saw that you added a `removeMemberSecret()` which might break that. I still need to review that part and see if there's a way to not have to use it)

Reply:
- done!
- `deepEqualJSONType` can catch proposals that are an exactly match, but I think we should be smarter than that. For example if I create one proposal to remove Bob and after if I create another one but with a different reason,  `deepEqualJSONType` wouldn't catch it. So that's why in the line 291-292 we check only for the `proposalData.member`. If it's a match, it's a duplicated, even if the reason for it is different. Does it make sense? Maybe we could compare the proposal data and exclude `reason` from it, so it's a generic "catch" for any proposal. What do you think?

Reply:
> Maybe we could compare the proposal data and exclude reason from it, so it's a generic "catch" for any proposal. What do you think?

@sandrina-p brilliant solution!

Reply:
Done!